### PR TITLE
New toolchain (2)

### PR DESCRIPTION
**Pull request by dinosaure**
This PR is #494 rebased according to `master` which includes #502 & #495. The only conflict is about the `configure.sh` and I just took the Martin's version. I would like to say that we mostly agreed (already) to merge this PR and cut a release. Our plan was postponed by the fact that if we want to continue to support MirageOS 3 or not.

Indeed, on top of this PR, @TheLortex did few patches to include the `pkg-config` path to pass C flags across packages. This PR **does not** include these patches and we agreed (the MirageOS core team) that we don't want to support MirageOS 3 and `pkg-config` stuffs at this layer (and on the `ocaml-freestanding` layer).

That mostly means that this PR can be merge as is and we can cut a release (`0.7.0`) then. However, I would like a triple check from the CI, @TheLortex and someone else from @mirage/core before merging this PR.

### CODE REVIEW

**Comment by hannesm** [comment]
I pushed https://github.com/dinosaure/solo5/pull/1 to have the configure.sh in shape with OpenBSD releases.

**Comment by dinosaure** [comment]
The patch was cherry-picked, thanks!

**Comment by adamsteen** [comment]
on a OpenBSD 7.0 Current, the following worked nicely:
`./configure.sh` && `gmake`

followed by `cd tests && doas ./setup-tests.sh && doas ./run-tests.sh`, output:

```
 ✓ noexec host
 ✓ hello hvt
 - hello virtio (skipped)
 - hello spt (skipped)
 - hello xen (skipped: xl not available)
 ✓ quiet hvt
 - quiet virtio (skipped)
 - quiet spt (skipped)
 ✓ globals hvt
 - globals virtio (skipped)
 - globals spt (skipped)
 - globals xen (skipped: xl not available)
 ✓ exception hvt
 - exception virtio (skipped)
 - exception spt (skipped)
 - exception xen (skipped: xl not available)
 ✓ zeropage hvt
 - zeropage virtio (skipped)
 - zeropage spt (skipped)
 - zeropage xen (skipped: xl not available)
 ✓ xnow hvt
 - xnow spt (skipped)
 - xnow xen (skipped: xl not available)
 ✓ wnox hvt
 - wnox spt (skipped)
 - wnox xen (skipped: xl not available)
 ✓ notls hvt
 - notls virtio (skipped)
 - notls spt (skipped)
 - notls xen (skipped: xl not available)
 - tls hvt (skipped: not supported on OpenBSD)
 - tls virtio (skipped)
 - tls spt (skipped)
 - tls xen (skipped: xl not available)
 ✓ ssp hvt
 - ssp virtio (skipped)
 - ssp spt (skipped)
 - ssp xen (skipped: xl not available)
 ✓ fpu hvt
 - fpu virtio (skipped)
 - fpu spt (skipped)
 - fpu xen (skipped: xl not available)
 ✓ time hvt
 - time virtio (skipped)
 - time spt (skipped)
 - time xen (skipped: xl not available)
 - seccomp spt (skipped)
 ✓ blk hvt
 - blk virtio (skipped)
 - blk spt (skipped)
 ✓ net hvt
 - net virtio (skipped)
 - net spt (skipped)
 - net_2if hvt (skipped: breaks on OpenBSD due to #374)
 - net_2if spt (skipped)
 - dumpcore hvt (skipped: not supported on OpenBSD)
 ✓ mft_maxdevices hvt
 - mft_maxdevices spt (skipped)

58 tests, 0 failures, 43 skipped
```

**Comment by dinosaure** [comment]
On Debian 10 with Xen:
```
root@solo5-test:~/solo5/tests# ./run-tests.sh 
 ✓ noexec host
 - hello hvt (skipped: no access to /dev/kvm or not present)
 ✓ hello virtio
 ✓ hello spt
 ✓ hello xen
 - quiet hvt (skipped: no access to /dev/kvm or not present)
 ✓ quiet virtio
 ✓ quiet spt
 - globals hvt (skipped: no access to /dev/kvm or not present)
 ✓ globals virtio
 ✓ globals spt
 ✓ globals xen
 - exception hvt (skipped: no access to /dev/kvm or not present)
 ✓ exception virtio
 ✓ exception spt
 ✓ exception xen
 - zeropage hvt (skipped: no access to /dev/kvm or not present)
 ✓ zeropage virtio
 ✓ zeropage spt
 ✓ zeropage xen
 - xnow hvt (skipped: no access to /dev/kvm or not present)
 ✓ xnow spt
 ✓ xnow xen
 - wnox hvt (skipped: no access to /dev/kvm or not present)
 ✓ wnox spt
 ✓ wnox xen
 - notls hvt (skipped: no access to /dev/kvm or not present)
 ✓ notls virtio
 ✓ notls spt
 ✓ notls xen
 - tls hvt (skipped: no access to /dev/kvm or not present)
 ✓ tls virtio
 ✓ tls spt
 ✓ tls xen
 - ssp hvt (skipped: no access to /dev/kvm or not present)
 ✓ ssp virtio
 ✓ ssp spt
 ✓ ssp xen
 - fpu hvt (skipped: no access to /dev/kvm or not present)
 ✓ fpu virtio
 ✓ fpu spt
 ✓ fpu xen
 - time hvt (skipped: no access to /dev/kvm or not present)
 ✓ time virtio
 ✓ time spt
 ✓ time xen
 ✓ seccomp spt
 - blk hvt (skipped: no access to /dev/kvm or not present)
 ✓ blk virtio
 ✓ blk spt
 - net hvt (skipped: no access to /dev/kvm or not present)
 ✓ net virtio
 ✓ net spt
 - net_2if hvt (skipped: no access to /dev/kvm or not present)
 ✓ net_2if spt
 - dumpcore hvt (skipped: no access to /dev/kvm or not present)
 - mft_maxdevices hvt (skipped: no access to /dev/kvm or not present)
 ✓ mft_maxdevices spt

58 tests, 0 failures, 17 skipped
```

**Comment by dinosaure** [comment]
On Debian 10 with KVM:
```
root@da-c3-small-x86-01:~/solo5/tests# ./run-tests.sh 
 ✓ noexec host
 ✓ hello hvt
 ✓ hello virtio
 ✓ hello spt
 - hello xen (skipped: xl not available)
 ✓ quiet hvt
 ✓ quiet virtio
 ✓ quiet spt
 ✓ globals hvt
 ✓ globals virtio
 ✓ globals spt
 - globals xen (skipped: xl not available)
 ✓ exception hvt
 ✓ exception virtio
 ✓ exception spt
 - exception xen (skipped: xl not available)
 ✓ zeropage hvt
 ✓ zeropage virtio
 ✓ zeropage spt
 - zeropage xen (skipped: xl not available)
 ✓ xnow hvt
 ✓ xnow spt
 - xnow xen (skipped: xl not available)
 - wnox hvt (skipped: not supported on Linux)
 ✓ wnox spt
 - wnox xen (skipped: xl not available)
 ✓ notls hvt
 ✓ notls virtio
 ✓ notls spt
 - notls xen (skipped: xl not available)
 ✓ tls hvt
 ✓ tls virtio
 ✓ tls spt
 - tls xen (skipped: xl not available)
 ✓ ssp hvt
 ✓ ssp virtio
 ✓ ssp spt
 - ssp xen (skipped: xl not available)
 ✓ fpu hvt
 ✓ fpu virtio
 ✓ fpu spt
 - fpu xen (skipped: xl not available)
 ✓ time hvt
 ✓ time virtio
 ✓ time spt
 - time xen (skipped: xl not available)
 ✓ seccomp spt
 ✓ blk hvt
 ✓ blk virtio
 ✓ blk spt
 ✓ net hvt
 ✓ net virtio
 ✓ net spt
 ✓ net_2if hvt
 ✓ net_2if spt
 ✓ dumpcore hvt
 ✓ mft_maxdevices hvt
 ✓ mft_maxdevices spt

58 tests, 0 failures, 12 skipped
```

File: scripts/opam-release.sh [hunk 1]
```diff
@@ -43,10 +43,11 @@ url {
   checksum: "sha512=${CHECKSUM}"
 }
 EOM
-for VARIANT in hvt spt virtio muen xen; do
-    PKG_DIR=${OUTPUT_DIR}/packages/solo5-bindings-${VARIANT}/solo5-bindings-${VARIANT}.${OPAM_VERSION}
+# Eventually we want '' 'cross-aarch64' ...
+for VARIANT in ''; do
```
Comment (hannesm) [inline scripts/opam-release.sh:47]:
```suggestion
for VARIANT in '' 'cross-aarch64'; do
```

File: scripts/opam-release.sh [hunk 2]
```diff
@@ -43,10 +43,11 @@ url {
   checksum: "sha512=${CHECKSUM}"
 }
 EOM
-for VARIANT in hvt spt virtio muen xen; do
-    PKG_DIR=${OUTPUT_DIR}/packages/solo5-bindings-${VARIANT}/solo5-bindings-${VARIANT}.${OPAM_VERSION}
+# Eventually we want '' 'cross-aarch64' ...
```
Comment (hannesm) [inline scripts/opam-release.sh:46]:
```suggestion
```

**Comment by dinosaure** [comment]
This PR seems fine for any of us and we did by hands tests needed on different platform which ensure that the next release of Solo5 will work everywhere. I think we are ready to cut a release. I will do first the process as explained into the `RELEASE.md` and then, send an email on the mailing list to announce this huge release. Thanks for all to participate on this PR!

**Pull request merged**